plugins/inputs/win_perf_counters/win_perf_counters.go

```diff
@@ -341,61 +406,79 @@ func (m *Win_PerfCounters) ParseConfig() error {
 
 }
 
-func (m *Win_PerfCounters) Gather(acc telegraf.Accumulator) error {
+func (m *WinPerfCounters) Gather(acc telegraf.Accumulator) error {
 	// Parse the config once
 	var err error
 
 	if m.lastRefreshed.IsZero() || (m.CountersRefreshInterval.Duration.Nanoseconds() > 0 && m.lastRefreshed.Add(m.CountersRefreshInterval.Duration).Before(time.Now())) {
-		if m.counters != nil {
-			m.counters = m.counters[:0]
-		}
-
-		if err = m.query.Open(); err != nil {
+		if err = m.cleanQueries(); err != nil {
```

Comment:
This seems to call a `.Close()` on the query, is that intended?

Reply:
In this branch, all previous counter infos must be cleaned and fresh infos are added.
There were only one reused query object previously, where `open` function called internally `close` if necessary.